### PR TITLE
update 1.6.1

### PR DESCRIPTION
**Pull request by Arishamays1**
## ☆ Snowflake-Snowpark-Python Update 1.6.1 (PKG-2577)☆
[Jira Ticket](https://anaconda.atlassian.net/browse/PKG-2577)
[Upstream](https://github.com/snowflakedb/snowpark-python)
[Dependencies ](https://github.com/snowflakedb/snowpark-python/blob/v1.6.1)
# Changes
- Updated version number and `sha256`
- Updated pinnings on dependencies

### CODE REVIEW

**Review by ViridianMelody — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -27,13 +27,15 @@ requirements:
     - cloudpickle >=1.6.0,<=2.0.0
     - snowflake-connector-python >=2.7.12,<4.0.0
```
Comment (cbouss) [inline recipe/meta.yaml:28]:
```suggestion
    - snowflake-connector-python >=3.0.4,<4.0.0
```
https://github.com/snowflakedb/snowpark-python/blob/v1.6.1/setup.py#L13

File: recipe/meta.yaml [hunk 2]
```diff
@@ -27,13 +27,15 @@ requirements:
...
     - typing-extensions >=4.1.0,<5.0.0
```
Comment (cbouss) [inline recipe/meta.yaml:29]:
```suggestion
```
Removed from **https://github.com/snowflakedb/snowpark-python/blob/v1.6.1/setup.py#L49**

File: recipe/meta.yaml [hunk 3]
```diff
@@ -27,13 +27,15 @@ requirements:
...
+    - pyaml
```
Comment (cbouss) [inline recipe/meta.yaml:30]:
```suggestion
    - pyyaml
```
https://github.com/snowflakedb/snowpark-python/blob/v1.6.1/setup.py#L54

**Review by cbouss — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 3 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: thank you!

**Pull request merged**